docs/args-and-options.md

```diff
@@ -0,0 +1,180 @@
+---
+id: args-and-options
+title: "Args and Options"
+---
+
+`Args[ArgsType]` and `Options[OptionsType]` represent CLI flags. They are descriptions of the process of constructing instances of `ArgsType` and `OptionsType` from a valid input.
+They can be used to construct a `Command[(ArgsType, OptionsType)]` using one of the following methods:
+```scala mdoc:silent
+import zio.cli._
+
+val name = "name"
+val optionsNone = Options.none
+val argsNone = Args.none
+
+Command(name)
+Command(name, optionsNone)
+Command(name, argsNone)
+Command(name, optionsNone, argsNone)
+```
+
+## Args
+`Args[A]` is a description of the process of constructing an instance of `A` from a valid input of the CLI. It is not yet a specified argument for the CLI. In other words, an `Args[A]` defines a collection of valid arguments for a command and a way to construct a value `A` from them.
+
+**ZIO CLI** offers a variety of methods to create basic `Args` that take a string from the user as input and transform it into a value of the corresponding type. 
+```scala mdoc:silent
+// Basic constructions
+Args.bool(name)
+Args.file(name)
+Args.directory(name)
+Args.text(name)
+Args.decimal(name)
+Args.integer(name)
+Args.instant(name)
+Args.localDate(name)
+Args.localDateTime(name)
+Args.localTime(name)
+Args.monthDay(name)
+Args.offsetDateTime(name)
+Args.offsetTime(name)
+Args.period(name)
+Args.year(name)
+Args.yearMonth(name)
+Args.zonedDateTime(name)
+Args.zoneId(name)
+Args.zoneOffset(name)
+```
+We can also avoid specifying a name of the `Args`. In this case, the name will be the type:
+```scala mdoc:silent
+Args.bool(name) // Boolean Args with name
+Args.bool       // Boolean Args named "boolean"
+```
+
+### Combining and transforming Args
+It is possible to create more complex `Args` using the following methods of the trait `Args`:
+```scala mdoc:silent
+trait Args[A] {
```

Comment:
I would divide the operators into categories, and then provide a section for each family of operators (e.g. "Repeating Arguments"), showing examples of each.